### DISCUSSION TITLE
How do I change the order of execution of code cells, inline code or sections in a Quarto single document or multiple documents in a project?

### DISCUSSION THREAD

**Discussion by fernando-dias** [quarto-dev/quarto-cli#10637] [Q&A]
### Description

Hi, Quarto newbie here. 

I think it is really common to see a few results in abstracts, or in the first pages of a presentation, so I wanted to add inline code into my Abstract that shows some numbers that are calculated elsewhere in my manuscript. I know that YAML abstract does not allow inline code, but there is an extension for that. I thought there would be a easy way to change the order of execution of chapters, sections, or documents to render. I have already checked the guide and searched for similar questions already discussed but they seem to try to solve more complex problems. The feature I am looking for seems to be kind of obvious for me, so I guess there is something quite easy or obvious that I am not seeing. I am using the last version of Quarto with Python code cells only. I know that Jupyter notebooks execute codes at the order they appear in the the text, an that OJS is more flexible, but I need to use Python.

Could you please help me?

**Comment by mcanouil** [comment]
@cscheid is that something the feature you merged recently can do?

**Reply by cscheid** [reply]
Maybe: https://prerelease.quarto.org/docs/authoring/contents.html

**Reply by fernandortdias** [reply]
"Sounds" nice to me. I'll try it in the next days. Thanx!

**Comment by fernando-dias** [comment]
Hi. 

I haven't tried it yet, but:

1. At the last cell of [documentation](https://prerelease.quarto.org/docs/authoring/contents.html ):

We will define some values here. [`{python} "some code"`]{#a-span}

The output of that inline code cell is . Etc.

Shouldn't include {{< contents a-span >}}, like the text below?

We will define some values here. [`{python} "some code"`]{#a-span}

The output of that inline code cell is {{< contents a-span >}}. Etc.

Moreover:

2. The second line int the example above could be somewhere before the first line in the (original) Quarto document?

3) The inline code [`{python} "some code"`]{#a-span} will be hidden in the rendered document?

**Reply by cscheid** [reply]
It's quite hard to understand what you're saying. Please give us a reproducible example of what you've attempted and how it doesn't work, and then we might be able to help you.

**Reply by cscheid** [reply]
> I haven't tried it yet

In the future, please try to use the feature before asking questions (and filing bug reports).

**Reply by fernando-dias** [reply]
Sorry for my confusion. For my second and third questions above, I'll try the prelease.

Reguarding my first question, that may be a issue in the documentation of the prerelease:

The last cell at https://prerelease.quarto.org/docs/authoring/contents.html 
has the following text:

Quote:

We will define some values here. [{python} "some code"]{#a-span}

The output of that inline code cell is . Etc.

Unquote.

I thought that "{{< contents a-span >}}" may be missing after "cell is" and before ". Etc." in the text above.

**Comment by natecostello** [comment]
My use case involves a lot of code, followed by markdown sections for the document, followed by an appendix that documents the code.  It would be incredibly useful to be able to embed or display code blocks from earlier in the document.  It sounds like the new Contents shortcode could be the solution.  My first attempts haven't been successful (even using the minimal example given).  I've submitted an issue separately.   This comment is just to express support for this functionality.
